### PR TITLE
Add arbo implementation for censustree interface

### PR DESCRIPTION
**Pull request by arnaucube**

### CODE REVIEW

File: censustree/arbotree/wrapper.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		return nil, err
+	}
+	tree := &Tree{
+		Tree: mt,
+	}
+	tree.updateAccessTime()
+	return tree, nil
+}
+
+func (t *Tree) Init(name, storageDir string) error {
```
Comment (p4u) [inline censustree/arbotree/wrapper.go:55]: Please add a description for `Init` (the text below might be added here).

File: censustree/arbotree/wrapper.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	mt, err := arbo.NewTree(database, 140, arbo.HashFunctionBlake2b)
+	if err != nil {
+		return err
+	}
+	t.Tree = mt
+	t.updateAccessTime()
+	return nil
+}
+
+func (t *Tree) MaxKeySize() int {
```
Comment (p4u) [inline censustree/arbotree/wrapper.go:73]: Add description to all Exported methods please.

File: censustree/factory/factory.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+const (
+	TreeTypeArboBlake2b = iota
+	TreeTypeArboPoseidon
+	TreeTypeGraviton
+)
+
+// NewCensusTree creates a merkle tree using the given storage and hash
+// function.  Note that each tree should use an entirely separate namespace for
+// its database keys.
+func NewCensusTree(typ int, name, storageDir string) (censustree.Tree, error) {
```
Comment (p4u) [inline censustree/factory/factory.go]: rename `typ` to something more self-descripting such as "treeType"

File: censustree/factory/factory.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	var err error
+	var tree censustree.Tree
+	switch typ {
+	case TreeTypeArboBlake2b:
+		tree, err = arbotree.NewTree(name, storageDir, arbo.HashFunctionBlake2b)
+		if err != nil {
```
Comment (p4u) [inline censustree/factory/factory.go]: You can use `if tree, err = ...`

**Review by p4u — APPROVED** [review]
LGTM, just minor comments (please address them before merging)

File: censustree/arbotree/wrapper.go (hunk 1 above)
Comment (arnaucube) [reply censustree/arbotree/wrapper.go:55]: Done, also added the descriptions to the methods of `censustree` & `graviton` packages.

File: censustree/arbotree/wrapper.go (hunk 2 above)
Comment (arnaucube) [reply censustree/arbotree/wrapper.go:73]: Done, also added the descriptions to the methods of `censustree` & `graviton` packages.

**Pull request merged**